ci/snap/ci-deploy.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+set -e -o pipefail -u -x
+
+sudo apt-get update
+sudo apt-get install -y snapd
+sudo snap install snapcraft --classic
+sudo lxd init --minimal
+# `--use-lxd` applied to use a LXD container instead of a VM, to work around lack of support for KVM on CircleCI VMs.
+snapcraft --use-lxd
+
+if [[ ${1-} == "--dry-run" || ${CIRCLE_BRANCH-} != "master" ]]; then
+  ! command -v git-machete
+  sudo snap install git-machete*.snap --dangerous --classic
+  git machete version
+  sudo snap remove git-machete
+  echo "MASTER"
+  echo "$SNAPCRAFT_LOGIN_CREDENTIALS_CONTENTS_BASE64" | base64 -d > ~/.snapcraft.login
```

Comment:
A security notice/warning here: while `$SNAPCRAFT_LOGIN_CREDENTIALS_CONTENTS_BASE64` will be masked in the CI logs (which are **public**), the un-base64-ed contents won't... AFAICS now they don't leak, but be wary of that (turning of `set -x` might be a good idea in the long run)